apis/network/v1beta1/reservation_types.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// IPAddress -
+type IPAddress struct {
+	// Network name
+	Network NetNameStr `json:"network"`
+
+	// Subnet name
+	Subnet NetNameStr `json:"subnet"`
+
+	// Address contains the IP address
+	Address string `json:"address"`
+}
+
+// ReservationSpec defines the desired state of Reservation
+type ReservationSpec struct {
+	// Hostname
+	Hostname *HostNameStr `json:"hostname"`
```

Comment:
We probably don't need this as we've IPSetRef?

Reply:
yes, could do that. I thought its easier to identify for any troubleshooting which host this is mapped to

Reply:
done, removed hostname